hazelcast-client-legacy/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -355,34 +357,33 @@ protected void addNearCacheInvalidateListener() {
         }
     }
 
+    protected final class InvalidationListener implements EventHandler<Object> {
 
-    protected static final class InvalidationListener implements EventHandler<PortableEntryEvent> {
-
-        protected final NearCache nearCache;
-
-        protected InvalidationListener(NearCache nearCache) {
-            this.nearCache = nearCache;
+        public InvalidationListener() {
         }
 
         @Override
-        public void handle(PortableEntryEvent event) {
-            switch (event.getEventType()) {
-                case ADDED:
-                case REMOVED:
-                case UPDATED:
-                case MERGED:
-                case EVICTED:
-                case EXPIRED:
-                    final Data key = event.getKey();
+        public void handle(Object event) {
+            if (event instanceof BatchNearCacheInvalidationData) {
```

Comment:
Do you need to check the source uuid if it is client itself originated or not?
(I have just asked because maybe you already filter it at server side)